app/controllers/email_authorizations_controller.rb

```diff
@@ -0,0 +1,12 @@
+class EmailAuthorizationsController < ApplicationController
+  def verify
+    user = User.find_by(username: params[:username])
+    raise ActionController::RoutingError, "Not Found" unless user && current_user == user
```

Comment:
Error handling here isn't super elegant, but I think that a 404 is best.

Reply:
Hmm maybe a `before_action :authenticate_user!` and `not_found unless current_user == user`. You could also make an `email_authorization_policy.rb` here if you want to go a step further.